### ISSUE TITLE
Conversion to USWDS v3 - Radio Button - Documentation

### ISSUE THREAD

**Issue by k80bowman** [department-of-veterans-affairs/vets-design-system-documentation#1269]
## Description
Update documentation and guidance for the Radio Button component on design.va.gov based on guidance from USWDS v3.

## Tasks
- [ ] Update component documentation for design.va.gov
- [ ] Request documentation draft review
- [ ] Once documentation is approved, publish to design.va.gov

## Acceptance Criteria
- [ ] Component documentation is published on design.va.gov

**Comment by humancompanion-usds** [comment]
I'm going to use Radio button as my prototype of how we'll reference USWDS guidance to fulfill issue #1200

**Comment by GnatalieH** [comment]
> I'm going to use Radio button as my prototype of how we'll reference USWDS guidance to fulfill issue #1200

Great idea.

**Comment by GnatalieH** [comment]
[Here is a link](https://www.sketch.com/s/d4382de7-dd0a-4ead-9cb1-14d6cafb587f/a/agLw0kp) to a mockup I'm working on for Radio button. 

After going through this exercise of comparing content between VADS and USWDS documentation sites, I have thoughts about how to proceed. Here are some considerations:

- The US system does not offer deep links to all of their subsections, only: Preview, Code, Guidance, Package, and Latest Updates. There is no deep link to the Accessibility subsection.
- We will want to differentiate between VA and USWDS components in the Examples section.
- In terms of Guidance, there will probably not be a need to differentiate between the VA and USWDS components because the guidance should be the same. Calling out the diffs here only would make things more confusing, IMO. 
- In many cases the VA documentation is already heavily cribbing from USWDS, verbatim, and I don't see the problem with copying over more content from USWDS when it's appropriate.
- Code usage section will probably need to have information that distinguishes between VA and USWDS components.
- Accessibility considerations probably won't need to mention diffs between the VA and US systems. Again, we are already heavily borrowing from the US system here and unless it's specific to the code, calling out VA vs. USWDS in this section could be confusing to the reader.
- Package information and Latest updates are unique subsections in USWDS documentation. We can place these as Action links under Code usage, or as text links under a Related section at the bottom of the page.

To summarize, I think we should look at documentation on a case-by-case basis when we are consuming a USWDS component. There is already a lot of overlap, including verbatim content, and simply copying a few more bullet points from the US system, where necessary, seems like the most intuitive path forward. We will likely always have additional content on the VA documentation site we can insert to supplement the overlapping content.

CC @humancompanion-usds

**Comment by caw310** [comment]
Carrying over to Sprint 19

**Comment by humancompanion-usds** [comment]
@GnatalieH - The issue is maintenance of that guidance with a small team. I'd rather not duplicate content because then we'd have to stay on top of diffs and do more copying and pasting across all components, which is a lot of work for a team that should be focused on writing new guidance. I agree that decisions should be made on a case-by-case basis, but I also believe we should have 3 categories of components and a guidance page design for each. Perhaps those variations are:

1. USWDS component & there is no additional VA guidance.
2. USWDS components & there is additional VA guidance.
3. VA component only (which is our current template).

For variation 2 I'd like to see what it looks like if we don't duplicate anything from USWDS. Let's use Radio button to do that and see where we end up. If it ends up that it is variation 1 then let's mock that up and next sprint we can find a different component to demo variation 2.

**Comment by GnatalieH** [comment]
[Here is the link](https://www.sketch.com/s/d4382de7-dd0a-4ead-9cb1-14d6cafb587f) to the final set of mockups that show all the variations @humancompanion-usds listed in his above comment.

On 1/19/23 Matt, Carol, Allison and I met to discuss the updated documentation and my mockups. We came to these conclusions:

- We should **not** present the USWDS component variant until 1-2 teams are piloting it. We _may_ put a call out on the page asking teams to contact the DST if they can pilot a component.
- The maturity badges, or something like them, could indicate when the latest version of component is USWDS v3. 
- We should indicate in the documentation which version we recommend, but also have content that supports teams who are using the older version(s).
- USWDS related props and attributes will get added to the table in Code usage. 
- We don't need action links to USWDS Package information or Code usage. N/A.
- We like the About section in the documentation and want to include it.

Is there anything I'm missing? 

I believe the next step should be to put these mockups on the back-burner until teams are piloting the v3 Radio button component. I'd like to try and pilot documentation with the component pilot. 

CC: @allison0034 and @caw310

**Comment by GnatalieH** [comment]
Hi @humancompanion-usds. [Here's a link](sketch.com/s/d4382de7-dd0a-4ead-9cb1-14d6cafb587f/a/pajqWwk) to my updated Radio button documentation. Here are some changes I made:

- Added an About section that I copied from USWDS
- Placed the USWDS example above the VA radio button example
- Moved Errors from the Usage section to Code usage
- Now Usage and Accessibility considerations sections just include an Action link back to the US system

Please let me know if this design looks good or not, thanks!

**Comment by humancompanion-usds** [comment]
I thought Allison's suggestion of adding a label to indicate if there is a USWDS variation was a good one. Some design element indicating what situation the component is in would be helpful. Also, it would be great if that could link to an explanation of the 3 variations that I defined above.

**Comment by GnatalieH** [comment]
@humancompanion-usds I'll add the label, I'd forgotten about that. I'm not sure I understand what you mean about linking to an explanation of the three variations, though.

**Comment by humancompanion-usds** [comment]
This works. I can use this for the Banner component that's up coming. The one thing we probably still need to tackle is a write up for teams on how/when they can use USWDS v3-based components. My plan is to add that to /components/ or somewhere in /about and link to it from the examples section with an action link. But I can add that as I prototype through it. We can close this.
